imagegw/shifter_imagegw/imagemngr.py

```diff
@@ -101,19 +99,58 @@ def __init__(self, config, logger=None, logname='imagemngr'):
         for system in self.config['Platforms']:
             self.systems.append(system)
         # Connect to database
-        if 'MongoDBURI' in self.config:
-            client = MongoClient(self.config['MongoDBURI'])
-            db_ = self.config['MongoDB']
-            self.images = client[db_].images
-        else:
+        if 'MongoDBURI' not in self.config:
             raise NameError('MongoDBURI not defined')
+        self.workers = WorkerThreads()
+        self.status_queue = self.workers.get_updater_queue()
+        self.status_proc = Process(target=self.status_thread,
+                                   name='StatusThread')
+        self.status_proc.start()
+        client = MongoClient(self.config['MongoDBURI'])
+        db_ = self.config['MongoDB']
+        self.images = client[db_].images
         self.metrics = None
         if 'Metrics' in self.config and self.config['Metrics'] is True:
             self.metrics = client[db_].metrics
-
-        initqueue(config)
         # Initialize data structures
 
+    def shutdown(self):
+        self.status_queue.put('stop')
+
+    def status_thread(self):
+        """
+        This listens for update messages from a queue.
+        """
+        client = MongoClient(self.config['MongoDBURI'])
+        db_ = self.config['MongoDB']
+        self.images = client[db_].images
```

Comment:
Lines 124-126 are the same as 109-111. You could consider adding a dedicated function to remove the duplication.